src/mlpack/methods/det/dtree_impl.hpp

```diff
@@ -165,10 +165,150 @@ DTree<MatType, TagType>::DTree() :
     right(NULL)
 { /* Nothing to do. */ }
 
+template <typename MatType, typename TagType>
+DTree<MatType, TagType>::DTree(const DTree& obj) :
+    start(obj.start),
+    end(obj.end),
+    maxVals(obj.maxVals),
+    minVals(obj.minVals),
+    splitDim(obj.splitDim),
+    splitValue(obj.splitValue),
+    logNegError(obj.logNegError),
+    subtreeLeavesLogNegError(obj.subtreeLeavesLogNegError),
+    subtreeLeaves(obj.subtreeLeaves),
+    root(obj.root),
+    ratio(obj.ratio),
+    logVolume(obj.logVolume),
+    bucketTag(obj.bucketTag),
+    alphaUpper(obj.alphaUpper)    
+{   
+    //Copying the children
+    if(obj.left != NULL)
+      left = new DTree(*obj.left);
+    else
+      left = NULL;
+    if(obj.right != NULL)
+      right = new DTree(*obj.right);
+    else
+      right = NULL;
```

Comment:
You can write these in the initialization list cleverly: `left((obj.left == NULL) ? NULL : new DTree(*obj.left))`, for instance.